gossip/privdata/pvtdataprovider.go

```diff
@@ -334,6 +346,10 @@ func (pdp *PvtdataProvider) populateFromTransientStore(pvtdata rwsetByKeys, pvtd
 					}
 					// populate the pvtdata with the RW set from the transient store
 					pvtdata[key] = col.Rwset
+					// decrement number of invalid eligible missing keys if writeset is invalid
+					if pvtdataRetrievalInfo.eligibleMissingKeys[key].invalid {
+						pvtdataRetrievalInfo.numInvalid--
+					}
```

Comment:
i paused here to think, why are we decrementing `numInvalid` when we find it to be invalid... 

if we do go down this approach, can we rename `numInvalid` to be `numSkip`, and comments could be more explicit
> decrement number of skipped key since invalid missing key is populated from cache/transient store

Reply:
> > I thought about that originally but decided against it since it seemed unintuitive that populateFromRemotePeers should return anything. I don't have any strong opinions though.
> 
> IMHO, it's naturally to expect some hints to be returned when it comes to "do something with remote peers". More importantly, i think the code would be more manageable if we don't carry struct member `numInvalid` all along, and alter it in both `populateFromCache` and `populateFromTransient`, **especially when we have all the info needed to compute such result in `populateFromRemotePeers`**.
> 
> here's what i have in mind [guoger@aa2cbf9](https://github.com/guoger/fabric/commit/aa2cbf9725bcda0de165f54fd587e62ff3cb18d5)
> 
> let me know what you think

I like it, this was the original approach I had intended so I'm fine with it if we're in agreement about returning the bool from `populateFromRemotePeers`

Reply:
Let me know if you want me to force push your commit on my branch or if you'd like to just make a new PR I'm fine with that too and we can close this one.

Reply:
@caod123 sorry for late response, please continue your PR here.